doc/source/whatsnew/v0.24.2.rst

```diff
@@ -22,13 +22,14 @@ Fixed Regressions
 
 - Fixed regression in :meth:`DataFrame.all` and :meth:`DataFrame.any` where ``bool_only=True`` was ignored (:issue:`25101`)
 
+- Fixed regression in :meth:`DataFrame.__init__`  where mixed type of rows generated `Segmentation fault` (:issue:`25075`)
```

Comment:
Fixed issue in ``DataFrame`` construction with passing a mixed list of mixed types could segfault.